### PR TITLE
log4net to Serilog: Log format cleanup / rework

### PR DESCRIPTION
**Pull request by tippmar-nr**
Fixed some issues with log formatting, particularly in the Audit log output. Added another Serilog package to help with defining the log format similarly to the way we did with log4net.

Modified the various `XxxFormat()` methods to pass the format string and args through to Serilog, which ensures proper formatting of the string especially when the format string contains embedded JSON.

Also re-worked the guard checks in `Logger.cs` and `Log.cs` so that the check is made at the lowest level.

### CODE REVIEW

**Review by chynesNR — APPROVED** [review]

**Comment by tippmar-nr** [comment]
Waiting on a green CI run before I merge. Not positive I've got all the edge cases handled yet.

**Review by jaffinito — APPROVED** [review]
Nice work here!

**Pull request merged**
